tests/NoUnused/ExportsTest.elm

```diff
@@ -1239,6 +1233,7 @@ unusedInProductionCode = 2
 module ATest exposing (..)
 import A
 import Test
+a : Test.Test
```

Comment:
Just to confirm: this is to avoid `a` from being reported in this test, right?

I'm slightly worried about the impact of this PR on tests. Some people don't add type annotations to their values, including their tests. As much as I like adding type annotations myself, I don't want to impose others to do so.

The fact that there is "by chance" no automatic fix here is nice because it won't cause the tests to disappear, which is a bit too easy  if the user does not pay much attention and will be hard to notice after the fact.

I think we should at least add a warning in the description asking the users to either add a type annotation if they want to keep a value that might be a test.

That said, I think we do want to have fixes for this, would you by any chance be up for adding that too? You're allowed to say no :smile:

The automatic fix would be removing the declaration and its potential `@docs` reference (which we already do for other exports), but let's skip the automatic if the module imports `Test` and the element is a value/function without a type annotation. Tests for this would be great.